tools/packaging/guest-image/build_image.sh

```diff
@@ -80,28 +86,43 @@ EOF
 }
 
 main() {
+	local host_oses=("cbl-mariner" "ubuntu")
```

Comment:
I think we will need to re-think the build of rootfs with marine as it is breaking the standard.

Currently, from a developer/CI point of view, the build process starts with a make command like:
```
$ rootfs-initrd-tarball
```

The kata-deploy's makefile targets are defined in `tools/packaging/kata-deploy/local-build/Makefile`. For rootfs builds, the target pattern name is `rootfs-(initrd|image)-[flavor]-tarball` where `flavor` can be anything. For example, for Confidential Containers we use `flavor` to build TEE-specifc rootfs, for example, AMD SEV (so `rootfs-initrd-sev-tarball` is the target) and others will come. In the past we used to build the guest OS for various OS (ubuntu, alpine, fedora...) however we hadn't covered on CI resulting on broken variants, then the community decide to keep on the ubuntu-based. That's why we don't have a `rootfs-initrd-fedora-tarball`, for example. However, IMHO, with cbl-mariner we would start building more OS variants as long as they are maintained and tested on CI. Okay, having said all that, I suggest to create a `rootfs-initrd-mariner-tarball` target so that a developer/CI can build it as:

```
$ make rootfs-initrd-mariner-tarball
```

Next you need to adapt the `tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh` script to accept the `--build=rootfs-initrd-mariner`. In particular, have a look at `install_initrd()` which the function that calls the rootfs builder script (tools/packaging/guest-image/build_image.sh), i.e., this own script should be changed to receive the `initrd_distro` and `initrd_distro_version`. It is the caller of `build_image.sh`  that should read the distro name and version from `versions.yaml`.

Reply:
FYI @fidencio

Reply:
I wonder if we also should make it more explicit the default ones (as in rootfs-initrd-alpine-tarball, and rootfs-image-ubuntu-tarball), or if this is a little bit too much.